File: C/OJ/Leetcode/2020-3/645_findErrorNums.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *findErrorNums(int *nums, int numsSize, int *returnSize)
{
    int *ans = (int *)malloc(sizeof(int) * 2);
    int a = 1;
    for (size_t i = 1; i < numsSize; i++)
    {
        a ^= i;
    }
    int b = nums[0];
    for (size_t i = 1; i < numsSize; i++)
    {
        b ^= nums[i];
    }
    ans[0] = b ^ a;
    ans[1] = b ^ a;
    return ans;
}

File: Python/OJ/20_04/645_findErrorNums.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        nums.sort()
        l = len(nums)
        a = 0
        for i in range(l):
            if nums[i] < i+1:
                return[nums[i], i+1]
            elif nums[i] > i+1:
                if a:
                    return[nums[i], a]
                else:
                    a = i+1
